### DISCUSSION TITLE
Is it possible to create project-specific snippets?

### DISCUSSION THREAD

**Discussion by jalexw** [zed-industries/zed#20647] [Feature Requests]
[The Snippets docs](https://zed.dev/docs/snippets) seems to indicate it is only possible through the home directory config folder at `~/.config/zed/snippets`?

**Comment by Edsol** [comment]
Same here! Like [settings.json](https://zed.dev/docs/configuring-zed#settings-files) file, I expected the project snippets to be merged as well.

**Comment by jpike88** [comment]
this one is huge. hope it gets some more attention.

**Comment by MaddyGuthridge** [comment]
I work as a teacher, and so I use a different collection of feedback snippets for each assignment I mark. I have a script that temporarily sets up project snippets within a project while I mark, then removes them before sending feedback to students. In order for me to use Zed for my work, it needs to support per-project snippets.

I hope this can be added soon.

**Comment by warrenjokinen** [comment]
_What a great idea to have per-project snippets!_

It might be a while before the team implements this(?), but I can think of two hacks:

 1) Try **clever names in 1 snippet file** that would then let you filter by project:

``` json
# settings.json
{
  "\n🤪  Hey! Thanks!... \n": {
    "prefix": "Thank you - fun",
    "body": ["Hey thanks! 🤪 \n"]
  },

  "\n🙏  Thank you... \n": {
    "prefix": "Thank you - serious",
    "body": ["Thank you for your contribution! 🙏 \n"]
  },

  "\n🙏  Thank you, generous... \n": {
    "prefix": "Thank you generous - serious",
    "body": ["Thank you for your generous contribution! 🙏 🙏 \n"]
    }
}
```
<img width="664" height="253" alt="ty" src="https://github.com/user-attachments/assets/55c982a1-255b-48f2-9442-44022ee48b43" />

 2) Use **multiple snippet files in your snippet folder**, like:
`snippets.json`, `P1snippets.json`, `P2snippets.json`, etc.
Then use Raycast (Mac), or just a little bit of shell scripting to load up the Project you want to use into the actual settings. json file.

_Hope that helps!_

**Reply by MaddyGuthridge** [reply]
Creating a temporary snippets file seems like a decent solution for my scripting use case, although it would be less ideal for many others. Thanks for the suggestion though -- I'll definitely use that in the mean time :)

**Comment by warrenjokinen** [comment]
I am looking at the Mac software "Raycast" to develop and manage snippets.
Then snippets everywhere, with the same interface, _in theory_.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
